BackEndApp/src/main/java/com/example/backendapp/Controller/PlantProfileController.java

```diff
@@ -9,31 +9,43 @@
 
 @RestController
 @RequiredArgsConstructor
-public class PlantProfileController {
+public class  PlantProfileController {
 
     private final PlantProfileService plantProfileService;
 
+    @CrossOrigin(origins = "http://localhost:3000")
+    @GetMapping("/PlantProfiles/{id}")
+    public PlantProfile fetchPlantProfileById(@PathVariable("id") Long plantProfileId)
+    {
+        PlantProfile plant = plantProfileService.fetchPlantProfileById(plantProfileId);
+        return plant;

```

Comment:
You can directly return plantProfileService.fetchPlantProfileById(plantProfileId).